plugins/builds/update.js

```diff
@@ -53,20 +54,23 @@ module.exports = () => ({
                         }
                         // Check permission against the pipeline
                         // @TODO implement this
-                    } else {
-                        switch (desiredStatus) {
-                        case 'SUCCESS':
-                        case 'FAILURE':
-                        case 'ABORTED':
-                            build.meta = request.payload.meta || {};
-                            build.endTime = (new Date()).toISOString();
-                            break;
-                        case 'RUNNING':
-                            build.startTime = (new Date()).toISOString();
-                            break;
-                        default:
-                            throw boom.badRequest(`Cannot update builds to ${desiredStatus}`);
-                        }
+                    }
+
+                    return eventFactory.get(build.eventId).then(event => ({ build, event }));
+                }).then(({ build, event }) => {
+                    switch (desiredStatus) {
+                    case 'SUCCESS':
+                    case 'FAILURE':
+                    case 'ABORTED':
+                        build.meta = request.payload.meta || {};
+                        event.meta = { ...event.meta, ...build.meta };
```

Comment:
Shouldn't the event meta only be updated on `SUCCESS` and `ABORTED`? Is there a use case for updating event meta when the build is `FAILURE`?

Reply:
Nevermind, after discussing with you I think it's more consistent/intuitive to update the event metadata where we update the build metadata.